ulmo/runs/SSL/llc_ssl_kin/llc_ssl_kin.py

```diff
@@ -0,0 +1,257 @@
+    
+    Returns:
+        args: (dict) dictionary of the arguments.
+    """
+    parser = argparse.ArgumentParser("argument for training.")
+    parser.add_argument("--opt_path", type=str, help="path of 'opt.json' file.")
+    parser.add_argument("--func_flag", type=str, 
+                        help="flags of the function to be execute: 'extract_curl', 'train', 'evaluate', 'umap'")
+    args = parser.parse_args()
+    
+    return args
+
+def main_train(opt_path: str):
+    # loading parameters json file
+    opt = Params(opt_path)
+    opt = option_preprocess(opt)
+
+    # build data loader
+    train_loader = modis_loader_v2(opt)
+
+    # build model and criterion
+    model, criterion = set_model(opt, cuda_use=opt.cuda_use)
+
+    # build optimizer
+    optimizer = set_optimizer(opt, model)
+    
+    # read 'user' and 'pin' for comet log
+    with open('/etc/comet-pin-volume/username', 'r') as f:
+        user = f.read()
+    
+    with open('/etc/comet-pin-volume/password', 'r') as f:
+        pin = f.read()
+        
+    # comet log
+    experiment = Experiment(
+            api_key=pin,
+            project_name="LLC_modis2012_curl", 
+            workspace=user,
+    )
+    experiment.log_parameters(opt.dict)
+    
+    # training routine
+    for epoch in trange(1, opt.epochs + 1):
+
+        adjust_learning_rate(opt, optimizer, epoch)
+
+        # train for one epoch
+        time1 = time.time()
+        loss = train_model(train_loader, model, criterion, optimizer, epoch, opt, cuda_use=opt.cuda_use)
+        time2 = time.time()
+        print('epoch {}, total time {:.2f}'.format(epoch, time2 - time1))
+        
+        # comet
+        experiment.log_metric('loss', loss, step=epoch)
+        experiment.log_metric('learning_rate', optimizer.param_groups[0]['lr'], step=epoch)
+        
+        if epoch % opt.save_freq == 0:
+            save_file = os.path.join(
+                opt.save_folder, 'ckpt_epoch_{epoch}.pth'.format(epoch=epoch))
+            save_model(model, optimizer, opt, epoch, save_file)
+
+    # save the last model
+    save_file = os.path.join(
+        opt.save_folder, 'last.pth')
+    save_model(model, optimizer, opt, opt.epochs, save_file)
+    
+def model_latents_extract(opt, modis_data, model_path, save_path, save_key):
+    """
+    This function is used to obtain the latents of the training data.
+    Args:
+        opt: (Parameters) parameters used to create the model.
+        modis_data: (numpy.array) modis_data used in the latents
+            extraction process.
+        model_path: (string) path of the model file. 
+        save_path: (string) path to save the extracted latents
+        save_key: (string) key of the h5py file for the latents.
+    """
+    model, _ = set_model(opt, cuda_use=opt.cuda_use)
+    model_dict = torch.load(model_path)
+    model.load_state_dict(model_dict['model'])
+    modis_data = np.repeat(modis_data, 3, axis=1)
```

Comment:
This is a bad use of memory.

See how I am doing it now in `modis_ssl_full`

Reply:
But we can keep as is here.

Reply:
Yes, here we replaced the old function with the model_latents_extract function from latents_extraction file.